lightning/src/ln/channel.rs

```diff
@@ -5896,6 +5909,8 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<(&'a K, u32)> for Channel<Signer>
 				fee_base_msat: Readable::read(reader)?,
 				fee_proportional_millionths: Readable::read(reader)?,
 				cltv_expiry_delta: Readable::read(reader)?,
+				counterparty_htlc_minimum_msat: Readable::read(reader)?,
+				holder_htlc_maximum_msat: Readable::read(reader)?,
```

Comment:
Right, this is tricky - so we can't add new fields in the serialization here as its not backwards/forwards compatible. We have to instead initialize dummy values here, then read them like we previously did and copy them here. We also should probably not store redundant data in our Channel storage and any fields we add to `CounterpartyForwardingInfo` we should drop from the main `Channel` struct.

Reply:
So it would be done like
```
let mut minimum_depth = None;
if ver == 1 {
	// Read the old serialization from version 0.0.98.
	minimum_depth = Some(Readable::read(reader)?);
} else {
	// Read the 4 bytes of backwards-compatibility data.
	let _dummy: u32 = Readable::read(reader)?;
}
```

Reply:
Yes, we'd want to keep the existing deserialization logic 1-for-1 and then just copy the field into its new location.

Reply:
So we would be removing any reference to `counterparty_max_htlc_value_in_flight_msat` & `counterparty_htlc_minimum_msat` in the Channel struct and impl's?